batch/batch/worker/flock.py

```diff
@@ -0,0 +1,54 @@
+import fcntl
+import os
+import argparse
+import subprocess as sp
+
+
+class Flock:
+    def __init__(self, path, nonblock=False):
+        self.path = os.path.abspath(path)
+        self.flock_flags = fcntl.LOCK_EX
+        self.fds = []
+
+        if nonblock:
+            self.flock_flags |= fcntl.LOCK_NB
```

Comment:
Unless you need this I think we should remove it, we don't seem to catch the associated exception anywhere.
```
In [4]: fcntl.flock(open('/tmp/foo'), fcntl.LOCK_EX | fcntl.LOCK_NB)                                                                                                                           
Traceback (most recent call last):
  File "<ipython-input-4-8375ce168aff>", line 1, in <module>
    fcntl.flock(open('/tmp/foo'), fcntl.LOCK_EX | fcntl.LOCK_NB)
BlockingIOError: [Errno 35] Resource temporarily unavailable
```

Reply:
We'd have to stick the blocking activity onto a thread pool as we do elsewhere. We could maybe use `LOCK_NB` to first check if we can lock the file without using a thread pool.

Reply:
I needed it for the garbage collector for the file cache. I'll remove it from here and figure out the correct thing then.